src/libraries/Microsoft.Extensions.Configuration.Binder/gen/ConfigurationBindingGenerator.Parser.cs

```diff
@@ -147,7 +149,7 @@ private static bool IsValidRootConfigType(ITypeSymbol? type)
                 {
                     // List<string> is used in generated code as a temp holder for formatting
                     // an error for config properties that don't map to object properties.
-                    _sourceGenSpec.TypeNamespaces.Add("System.Collections.Generic");
+                    _sourceGenSpec.Namespaces.Add("System.Collections.Generic");
```

Comment:
The `record` type `InvocationDiagnosticInfo` at the start of this file also has broken equality semantics. Can't comment on it directly.

Reply:
Thanks, I have pending 9.0 work to fix this https://github.com/dotnet/runtime/pull/89587.